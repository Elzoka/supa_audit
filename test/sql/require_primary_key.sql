begin;

    create table public.dummy(
        id int
    );


    -- Should raise exception that there is no primary key
    select audit.enable_tracking('public.dummy');

rollback;
