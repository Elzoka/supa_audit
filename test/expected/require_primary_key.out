begin;
    create table public.dummy(
        id int
    );
    -- Should raise exception that there is no primary key
    select audit.enable_tracking('public.dummy');
ERROR:  Table dummy can not be audited because it has no primary key
CONTEXT:  PL/pgSQL function audit.enable_tracking(regclass) line 14 at RAISE
rollback;
